File: ExpenseTracker.Domain/Entities/Budget.cs

namespace ExpenseTracker.Domain.Entities;

public class Budget
{
    public int BudgetId { get; set; }

    public int UserId { get; set; }

    public int CategoryId { get; set; }

    public decimal Amount { get; set; }

    public int Month { get; set; }

    public int Year { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public User User { get; set; } = null!;

    public Category Category { get; set; } = null!;
}

File: expense-tracker-web/src/types/budget.ts
export interface Budget {
    budgetId: number;
    categoryId: number;
    categoryName: string;
    amount: number;
    month: number;
    year: number;
    spentAmount: number;
    remainingAmount: number;
    progressPercentage: number;
    createdAt: string;
    updatedAt: string;
}

export interface CreateBudgetRequest {
  categoryId: number;
  amount: number;
  month: number;
  year: number;
}

export interface UpdateBudgetRequest {
  amount: number;
}